pkg/platform/kube/functionres/lazy.go

```diff
@@ -2029,6 +2030,28 @@ func (lc *lazyClient) getMetricResourceByName(resourceName string) v1.ResourceNa
 	}
 }
 
+func (lc *lazyClient) getFunctionServiceType(function *nuclioio.NuclioFunction) (v1.ServiceType, error) {
+	functionHTTPTriggers := functionconfig.GetTriggersByKind(function.Spec.Triggers, "http")
+	if len(functionHTTPTriggers) != 1 {
```

Comment:
why is the validation occur here? move it to the early stage where all validations performed